src/server/dflycmd.cc

```diff
@@ -325,8 +325,8 @@ OpStatus DflyCmd::StartFullSyncInThread(FlowInfo* flow, Context* cntx, EngineSha
   };
 
   // Shard can be null for io thread.
+  CHECK(!sf_->journal()->OpenInThread(false, ""sv));  // can only happen in persistent mode.
```

Comment:
I had a bug that the shard unique count was not calculated in case the transaction was on non shard thread.
SetMultiUniqueShardCount is not called if ServerState::tlocal()->journal() = null